### PR TITLE
Maximum Latitude of a Great Circle Arc

### LINKED ISSUE #511: The maximum latitude for a GCA
**Describe the feature you'd like added to this project**
Attain the maximum latitude for a great circle arc

**Link to original functionality implementation elsewhere, if applicable**

**Links to other examples, if applicable**

**Describe the solution you'd like**

**Additional context**

### PR DESCRIPTION
**Pull request by hongyuchen1030**
Closes #511 

## Overview

- Added a new function `extreme_gca_latitude` to calculate the extreme (minimum or maximum) latitude of a great circle arc.
- The function's testcases utilizes an iterative approach to refine the calculation to a specified tolerance level.

## Expected Usage

```Python
import uxarray as ux

grid_path = "/path/to/grid.nc"
data_path = "/path/to/data.nc"

uxds = ux.open_dataset(grid_path, data_path)

# Calculate the maximum latitude of a great circle arc
max_latitude = uxds.extreme_gca_latitude(gca_cart, extreme_type='max')

# Calculate the minimum latitude of a great circle arc
min_latitude = uxds.extreme_gca_latitude(gca_cart, extreme_type='min')


```

## PR Checklist

**General**
- [x] An issue is linked created and linked
- [x] Add appropriate labels
- [x] Filled out Overview and Expected Usage (if applicable) sections

**Testing**
- [x] Adequate tests are created if there is new functionality
- [x] Tests cover all possible logical paths in your function
- [x] Tests are not too basic (such as simply calling a function and nothing else)

**Documentation**
- [x] Docstrings have been added to all new functions
- [x] Docstrings have updated with any function changes
- [x] Internal functions have a preceding underscore (`_`) and have been added to `docs/internal_api/index.rst`
- [x] User functions have been added to `docs/user_api/index.rst`

### CODE REVIEW

File: test/test_geometry.py [hunk 1]
```diff
@@ -153,3 +155,231 @@ def test_pole_point_inside_polygon_from_vertice_cross(self):
...
+
+    def test_extreme_gca_latitude_min(self):
+        # Define a great circle arc that is symmetrical around 0 degrees longitude
+        gca_cart = np.array([
+            ux.grid.coordinates.normalize_in_place([0.5, 0.5, -0.5]),
+            ux.grid.coordinates.normalize_in_place([-0.5, 0.5, -0.5])
+        ])
+
+        # Calculate the maximum latitude
+        min_latitude = ux.grid.lines.extreme_gca_latitude(gca_cart, 'min')
```
Comment (rajeeja) [inline test/test_geometry.py:366]: did you mean `min` - # Check if the maximum latitude is correct

`_min_latitude_rad_iterative `

**Review by rajeeja — APPROVED** [review]

File: test/test_geometry.py (hunk 1 above)
Comment (hongyuchen1030) [reply test/test_geometry.py:366]: Yes

**Comment by philipc2** [comment]
@hongyuchen1030 

Once you get these changes in, we can get this merged for the November release. Let me know if you need any help!

**Comment by hongyuchen1030** [comment]
> @hongyuchen1030
> 
> Once you get these changes in, we can get this merged for the November release. Let me know if you need any help!

I wonder how's the progress for #546 ? Is that possible that it can be merged in the near futrue

File: uxarray/grid/arcs.py [hunk 2]
```diff
@@ -150,3 +155,58 @@ def _angle_of_2_vectors(u, v):
     angle_u_v_rad = 2 * np.arctan2(np.linalg.norm(vec_minus),
                                    np.linalg.norm(vec_sum))
     return angle_u_v_rad
+
+
+def extreme_gca_latitude(gca_cart, extreme_type):
```
[the remark this answers is not in the document]
Comment (hongyuchen1030) [reply uxarray/grid/arcs.py:160]: I think it's better to expose this function as a user API since it can provide very helpful information about each arc

**Comment by philipc2** [comment]
> > @hongyuchen1030
> > Once you get these changes in, we can get this merged for the November release. Let me know if you need any help!
> 
> I wonder how's the progress for #546 ? Is that possible that it can be merged in the near futrue

Aiming to wrap it up this week!

File: uxarray/grid/arcs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (philipc2) [reply uxarray/grid/arcs.py:155]: That's fine by me.

**Review by philipc2 — APPROVED** [review]
Nice work!

**Pull request merged**